### PR TITLE
New Package: awesome-LuaJIT

### PR DESCRIPTION
**Pull request by mustaqimM**
This seems to work fine as I'm currently using it:
```zsh
$ awesome -v
awesome v4.3 (Too long)
 • Compiled against Lua 5.1.5 (running with LuaJIT 2.1.0-beta3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
```

### CODE REVIEW

**Comment by Piraty** [comment]
I think it makes more sense to extend the existing awesome template and make this a build_option, instead of adding another template.

**Comment by ndgnuh** [comment]
I don't think `lgi-luaJIT` is conflicted with `lgi`, since both use different version of lua and have different directory structure, except for the `/usr/bin/dump-typelib`, which is a `.lua` file, which can be renamed to `dump-typelib-luajit` or something like that
```
$ xbps-query --list-pkgs | grep lgi
ii lgi-0.9.2_2                            Lua binadings for gnome/gobject using gobject-introspection library
ii lgi-LuaJIT-0.9.2_1                     LuaJIT bindings for gnome/gobject using gobject-introspection library
.
```

File: srcpkgs/lgi-LuaJIT/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+makedepends="libglib-devel libffi-devel lua51-devel
+ gobject-introspection gir-freedesktop"
+depends="LuaJIT gir-freedesktop"
+short_desc="LuaJIT bindings for gnome/gobject using gobject-introspection library"
+maintainer="mustaqim <mustaqim@pm.me>"
+license="MIT"
+homepage="https://github.com/pavouk/lgi"
+distfiles="https://github.com/pavouk/lgi/archive/$version.tar.gz"
+checksum=cfc4105482b4730b3a40097c9d9e7e35c46df2fb255370bdeb2f45a886548c4f
+conflicts="lgi"
```
Comment (ndgnuh) [inline srcpkgs/lgi-LuaJIT/template]: Since `lgi` use `lua53` by default, this might not be necessary.

File: srcpkgs/lgi-LuaJIT/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'lgi-LuaJIT'
+pkgname=lgi-LuaJIT
```
Comment (ndgnuh) [inline srcpkgs/lgi-LuaJIT/template]: I think this package actually doesn't have anything to do with `LuaJIT`. Both `LuaJIT` and `lua51` can use what ever contained in that package. So I suggest changing the name to `lua51-lgi`.

**Comment by q66** [comment]
oh, it's an old PR. If nobody is interested in doing anything to this, we can close it

File: srcpkgs/lgi-LuaJIT/template [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+short_desc="LuaJIT bindings for gnome/gobject using gobject-introspection library"
+maintainer="mustaqim <mustaqim@pm.me>"
+license="MIT"
+homepage="https://github.com/pavouk/lgi"
+distfiles="https://github.com/pavouk/lgi/archive/$version.tar.gz"
+checksum=cfc4105482b4730b3a40097c9d9e7e35c46df2fb255370bdeb2f45a886548c4f
+conflicts="lgi"
+
+post_install() {
+	vbin tools/dump-typelib.lua dump-typelib
```
Comment (ndgnuh) [inline srcpkgs/lgi-LuaJIT/template]: For this one, I inspected the original `lgi` and this package. The only file that is conflict is the `tools/dump-typelib.lua`. Maybe we should change this to `dump-tpyelib5.1`, the one in the original `lgi` to `dump-typelib5.3` and add and alternative group?

**Comment by ndgnuh** [comment]
I'm working on a build option with awesome, but that will require some changes on lgi too. I'll send a PR soon.

**Comment by hippi777** [comment]
hi folks! :)

@q66, luajit is alive, there was no sign about Mike (its main developer) for something like a year, and he announced before that, that he has other duties, but some months ago he reappeared and started to work on luajit again. in the meantime, moonjit appeared, and its main purpose is to be a drop in replacement for luajit, and basically it collects the pending pull requests for luajit and makes releases more often. (< in case of interest...) however luajit is actually under development now, just no fresh release came out for a long while.

btw i dunno why its written formally, as LuaJIT, instead of luajit in the package name, but im a new kid here... :D

otherwise +1 for a build option instead, but ive got no interest in this...

bests! :)

**Comment by ndgnuh** [comment]
Hello, I've opened another PR #20443. Can anyone review please?

**Comment by q66** [comment]
@hippi777 having a fair amount of experience with LuaJIT, I don't see it as entirely sustainable considering the code is rather unorthodox and barely maintainable for anybody who isn't Mike, and Mike being around or not being around makes a bus factor of 1, so I'm wary of this either way. While moonjit might be a thing now, that's pretty much just a dumping grounds for unmerged LuaJIT pull requests, which is hardly an ideal solution in itself.

Anyway, closing this.

**Pull request closed without merging**